packages/next/src/build/swc/index.ts

```diff
@@ -159,15 +159,27 @@ let lastNativeBindingsLoadErrorCode:
   | 'unsupported_target'
   | string
   | undefined = undefined
+// Used to cache calls to `loadBindings`
+let pendingBindings: Promise<Binding>
+// some things call `loadNative` directly instead of `loadBindings`... Cache calls to that
+// separately.
 let nativeBindings: Binding
+// can allow hacky sync access to bindings for loadBindingsSync
 let wasmBindings: Binding
 let downloadWasmPromise: any
-let pendingBindings: any
 let swcTraceFlushGuard: any
 let downloadNativeBindingsPromise: Promise<void> | undefined = undefined
 
 export const lockfilePatchPromise: { cur?: Promise<void> } = {}
 
+/**
+ * Attempts to load a native or wasm binding.
+ *
+ * By default, this first tries to use a native binding, falling back to a wasm binding if that
+ * fails.
+ *
+ * This function is `async` as wasm requires an asynchronous import in browsers.
```

Comment:
is the browser behavior relevant? this doesn't ever get used browser side
```suggestion
 * This function is `async` as wasm requires an asynchronous import.
 ```

Reply:
We build and publish both "nodejs" and "web" targets: https://rustwasm.github.io/docs/wasm-pack/commands/build.html#target

The web target is needed for things like webcontainers (i.e. stackblitz).

Sidenote: this seems pretty annoying as AFAICT, the wasm binary blobs are basically the same and just the JS shim is different, so it doesn't make sense why these need to be entirely different packages

Reply:
ah TIL! ignore this then